src/DynamoCore/Graph/Nodes/NodeModel.cs

```diff
@@ -1348,6 +1348,21 @@ protected NodeModel()
             OutPorts.CollectionChanged += PortsCollectionChanged;
         }
 
+        private void DisposePort(PortModel portModel, bool nodeDisposing = false)
+        {
+            portModel.PropertyChanged -= OnPortPropertyChanged;
+            portModel.ConnectorCollectionChanged -= ConnectorsCollectionChanged;
+
+            // if this node is being disposed, we don't need to set node state and destroy the connectors,
+            // as the connectors will be deleted elsewhere
+            if (!nodeDisposing)
+            {
+                portModel.DestroyConnectors();
+                SetNodeStateBasedOnConnectionAndDefaults();
+            }
+            portModel.Dispose();
+        }
```

Comment:
This would look cleaner if this logic is refactored in the `PortModel.Dispose` method somehow.